spec/services/frameworks/question_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Frameworks::Question do
+  describe '#call' do
+    let(:fixture_path) { 'spec/fixtures/files/frameworks/person-escort-record-1/questions' }
+
+    it 'sets a question as required if required validation available' do
+      filepath = Rails.root.join(
+        "#{fixture_path}/medical-details-information.yml",
+      )
+      question = FrameworkQuestion.new(section: 'health', key: 'medical-details-information')
+      described_class.new(filepath: filepath, questions: { 'medical-details-information' => question }).call
+
+      expect(question.required).to eq(true)
+    end
+
+    it 'sets the type on a question' do
+      filepath = Rails.root.join(
+        "#{fixture_path}/medical-details-information.yml",
+      )
+      question = FrameworkQuestion.new(section: 'health', key: 'medical-details-information')
+      described_class.new(filepath: filepath, questions: { 'medical-details-information' => question }).call
+
+      expect(question.question_type).to eq('textarea')
+    end
+
+    it 'sets the options on a question' do
+      filepath = Rails.root.join(
+        "#{fixture_path}/regular-medication.yml",
+      )
+      question = FrameworkQuestion.new(section: 'health', key: 'regular-medication')
+      described_class.new(filepath: filepath, questions: { 'regular-medication' => question }).call
+
+      expect(question.options).to contain_exactly('Yes', 'No')
+    end
+
+    it 'does not set the options on a question if type does not permit it' do
+      filepath = Rails.root.join(
+        "#{fixture_path}/medical-details-information.yml",
+      )
+      question = FrameworkQuestion.new(section: 'health', key: 'medical-details-information')
+      described_class.new(filepath: filepath, questions: { 'medical-details-information' => question }).call
+
+      expect(question.options).to be_empty
+    end
+
+    it 'allows folllowup comments to be set on a question' do
```

Comment:
follow

Reply:
not sure I understand? is this referring to the `followup` in the block?

Reply:
No, just a spelling mistake - three l's!